### PR TITLE
Fix Mix_LoadWAV with SDL2 RWops

### PR DESCRIPTION
**Pull request by Daft-Freak**

### CODE REVIEW

**Comment by kripken** [comment]
I guess this is for the case where a RWops was created explicitly in C code? Good to fix this.

How about refactoring that code into a shared function, as it could be used by `LoadMUS` as well?

We should also add a test for this.

**Comment by Daft-Freak** [comment]
I've just realised that this will break the non-RWops versions of LoadWAV/MUS.

**Comment by Daft-Freak** [comment]
Yes, this is for RWops created explicitly. I've re-enabled the testing disabled by d7f203dd5cb7f763f686f4660a7a1446c0e185c6.

File: src/library_sdl.js [hunk 1]
```diff
@@ -2571,6 +2571,26 @@ var LibrarySDL = {
   Mix_LoadWAV_RW: function(rwopsID, freesrc) {
     var rwops = SDL.rwops[rwopsID];
 
+#if USE_SDL == 2
+    if (rwops === undefined) {
+      var type = {{{ makeGetValue('rwopsID + ' + 20 /*type*/, '0', 'i32') }}};
+
+      if(type == 2/*SDL_RWOPS_STDFILE*/) {
```
Comment (kripken) [inline src/library_sdl.js]: space after `if`, and please use `===` (triple)

File: src/library_sdl.js [hunk 2]
```diff
@@ -2571,6 +2571,26 @@ var LibrarySDL = {
...
+        var fp = {{{ makeGetValue('rwopsID + ' + 28 /*hidden.stdio.fp*/, '0', 'i32') }}};
+        var stream = FS.getStreamFromPtr(fp);
+        if (stream) {
+          rwops = {filename: stream.path};
```
Comment (kripken) [inline src/library_sdl.js]: space after `{` and before `}`.

File: src/library_sdl.js [hunk 3]
```diff
@@ -2571,6 +2571,26 @@ var LibrarySDL = {
...
+        }
+      }
+      else if(type == 4/*SDL_RWOPS_MEMORY*/ || type == 5/*SDL_RWOPS_MEMORY_RO*/) {
```
Comment (kripken) [inline src/library_sdl.js]: space after `if`

File: src/library_sdl.js [hunk 4]
```diff
@@ -2571,6 +2571,26 @@ var LibrarySDL = {
...
+        var base = {{{ makeGetValue('rwopsID + ' + 24 /*hidden.mem.base*/, '0', 'i32') }}};
+        var stop = {{{ makeGetValue('rwopsID + ' + 32 /*hidden.mem.stop*/, '0', 'i32') }}};
+
+        rwops = {bytes: base, count: stop - base};
```
Comment (kripken) [inline src/library_sdl.js]: same as above

**Comment by kripken** [comment]
Some style nits, otherwise looks good.

**Comment by Daft-Freak** [comment]
Done.

**Comment by kripken** [comment]
Thanks. Please just squash the commits into one.

**Comment by Daft-Freak** [comment]
Done.

**Comment by kripken** [comment]
Thanks!

**Pull request merged**